pkg/replication/replication.go

```diff
@@ -363,39 +363,50 @@ func (r *replication) handleReplicationEnd(session kp.Session, renewalErrCh chan
 	}
 }
 
-// note: logging should be delegated somehow
-func (r *replication) updateOne(
-	node types.NodeName,
-	done chan<- nodeUpdated,
-	timeoutCh <-chan struct{},
-	aggregateHealth *podHealth,
-) error {
-	targetSHA, _ := r.manifest.SHA()
-	nodeLogger := r.logger.SubLogger(logrus.Fields{"node": node})
-	nodeLogger.WithField("sha", targetSHA).Infoln("Updating node")
-
+func (r *replication) shouldScheduleForNode(node types.NodeName, logger logging.Logger) bool {
 	nodeReality, err := r.queryReality(node)
 	if err != nil || nodeReality == nil {
-		nodeLogger.WithError(err).Errorln("Could not read Reality for this node. Will proceed to schedule onto it.")
+		logger.WithError(err).Errorln("Could not read Reality for this node. Will proceed to schedule onto it.")
```

Comment:
not related to your code, but I think it's better to check `err == pods.NoCurrentManifest` rather than `nodeReality == nil` and don't log the error, (although an `Infoln()` would probably be useful)